voice_insights_metrics.go

```diff
@@ -0,0 +1,84 @@
+package twilio
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+const MetricsPathPart = "Metrics"
+
+type CallMetricsService struct {
+	callSid string
+	client  *Client
+}
+
+type CallMetricsPage struct {
+	Metrics []CallMetric `json:"metrics"`
+	Meta    Meta         `json:"meta"`
+}
+
+type CallMetric struct {
+	AccountSid  string       `json:"account_sid"`
+	CallSid     string       `json:"call_sid"`
+	CarrierEdge *EdgeMetrics `json:"carrier_edge,omitempty"`
+	ClientEdge  *EdgeMetrics `json:"client_edge,omitempty"`
+	Direction   string       `json:"direction"`
+	Edge        string       `json:"edge"`
+	SdkEdge     *EdgeMetrics `json:"sdk_edge,omitempty"`
```

Comment:
Hmm yes, with Voice Insights API, the existence of the various `edge` details in the Twilio payload depends on the call type. It would look nicer if someone serializes the struct directly from the library. What are your thoughts?

Reply:
Oh, right, this is more about people going from the struct to JSON, then the reverse. OK, I'm fine either way.